### DISCUSSION TITLE
Please add canSeeWhenAny(["...", "..."]) to ProxiesCanSeeToGate

### DISCUSSION THREAD

**Discussion by puzzledmonkey** [laravel/nova-issues#5310] [Q&A]
ProxiesCanSeeToGate allows us to use Field::make(...)->canSeeWhen('permission'); and it can also be used with arrays (even though the jsdoc says it should be a string only - Illuminate\Foundation\Auth\Access\Authorizable allows iterable|string for the first parameter).

This greatly cuts down the amount of boilerplate code required to do ->canSee(function($request) { return $request->user()->can(...); }) and is really useful.

Would it be possible to add another method - canSeeWhenAny(["...", "..."]) which just proxies to user()->canAny(...) instead of user()->can(...)? That will help in cases where multiple access permissions could allow access to the same card, dashboard, etc etc.

Thanks for considering! :)

**Comment by crynobone** [comment]
You can add macros to `Field`.

```php
Field::macro('canSeeWhenAny', function () {
  // ....
});
```

https://tighten.com/insights/the-magic-of-laravel-macros/

**Reply by puzzledmonkey** [reply]
That's great, but you can't do that to lenses, dashboards, etc? But that is a very useful feature nonetheless thank you.

(edit - i see Actions are macroable though, so that should work)
